src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/FindAllReferences/FindAllReferencesEndpoint.cs

```diff
@@ -0,0 +1,157 @@
+    {
+        private readonly LanguageServerFeatureOptions _featureOptions;
+        private readonly RazorDocumentMappingService _documentMappingService;
+
+        public FindAllReferencesEndpoint(
+            LanguageServerFeatureOptions languageServerFeatureOptions,
+            RazorDocumentMappingService documentMappingService,
+            ClientNotifierServiceBase languageServer,
+            ILoggerFactory loggerFactory,
+            LanguageServerFeatureOptions featureOptions,
+            TagHelperFactsService tagHelperFactsService,
+            HtmlFactsService htmlFactsService)
+            : base(languageServerFeatureOptions, documentMappingService, languageServer, loggerFactory.CreateLogger<FindAllReferencesEndpoint>())
+        {
+            _featureOptions = featureOptions;
+            _documentMappingService = documentMappingService ?? throw new ArgumentNullException(nameof(documentMappingService));
+        }
+
+        public RegistrationExtensionResult GetRegistration(VSInternalClientCapabilities clientCapabilities)
+        {
+            const string AssociatedServerCapability = "referencesProvider";
+
+            var registrationOptions = new ReferenceOptions()
+            {
+                // TODO: GH issue link to support progress 
+                WorkDoneProgress = false,
+            };
+
+            return new RegistrationExtensionResult(AssociatedServerCapability, new SumType<bool, ReferenceOptions>(registrationOptions));
+        }
+
+        protected override string CustomMessageTarget => RazorLanguageServerCustomMessageTargets.RazorReferencesEndpointName;
+
+        protected override Task<IDelegatedParams?> CreateDelegatedParamsAsync(ReferenceParamsBridge request, RazorRequestContext requestContext, Projection projection, CancellationToken cancellationToken)
+        {
+            var documentContext = requestContext.GetRequiredDocumentContext();
+            return Task.FromResult<IDelegatedParams?>(new DelegatedPositionParams(
+                    documentContext.Identifier,
+                    projection.Position,
+                    projection.LanguageKind));
+        }
+
+        protected override async Task<VSInternalReferenceItem[]> HandleDelegatedResponseAsync(VSInternalReferenceItem[] delegatedResponse, ReferenceParamsBridge originalRequest, RazorRequestContext requestContext, Projection projection, CancellationToken cancellationToken)
+        {
+            var remappedLocations = new List<VSInternalReferenceItem>();
+
+            foreach (var referenceItem in delegatedResponse)
+            {
+                if (referenceItem?.Location is null || referenceItem.Text is null)
+                {
+                    continue;
+                }
+
+                // Temporary fix for codebehind leaking through
+                // Revert when https://github.com/dotnet/aspnetcore/issues/22512 is resolved
```

Comment:
I was curious what this was, so I clicked it. Even though github redirects, would be good to update to point to https://github.com/dotnet/razor/issues/4611 now that its moved, so it might come up in a search

Reply:
Yea, I followed the thread and it seems like no work is planned until we push for it